internal/observability/middleware_test.go

```diff
@@ -1323,44 +1333,61 @@ func TestStreamingMetrics(t *testing.T) {
 
 				stream, err := transport.NewServerStream(&fakeStream{request: req})
 				require.NoError(t, err)
-				err = mw.HandleStream(stream, &fakeHandler{err: tt.err})
-				require.Error(t, err)
 
-				snap := root.Snapshot()
-				successTags := newTags(_directionInbound, "")
-				errTags := newTags(_directionInbound, tt.errName)
-
-				// so we don't have create a sorting implementaion, manually place the
-				// first two expected counter snapshots, based on the error fault.
-				counters := make([]metrics.Snapshot, 0, 10)
-				if statusFault(yarpcerrors.FromError(tt.err)) == clientFault {
-					counters = append(counters,
-						metrics.Snapshot{Name: "caller_failures", Tags: errTags, Value: 1},
-						metrics.Snapshot{Name: "calls", Tags: successTags, Value: 1})
-
-				} else {
-					counters = append(counters,
-						metrics.Snapshot{Name: "calls", Tags: successTags, Value: 1},
-						metrics.Snapshot{Name: "server_failures", Tags: errTags, Value: 1})
+				handleStreamAndValidateFn := func() {
```

Comment:
Augmenting this existing test makes it difficult to see what the exact behaviour is, especially since we have three different `if tt.panics` checks.

Can you separate the panic test into its own subtest, even if it is largely a copy of the original test?